include/toml++/impl/preprocessor.h

```diff
@@ -399,7 +399,27 @@
 
 // dllexport etc
 #ifndef TOML_API
-	#define TOML_API
+	#if TOML_HEADER_ONLY == 0
+		#if defined(_WIN32) || defined(__CYGWIN__)
+			#ifdef TOML_SHARED_LIB
+				#ifdef TOML_BUILDING_LIB
```

Comment:
```suggestion
			#if TOML_SHARED_LIB
				#if TOML_SHARED_LIB_BUILDING
```

I don't like just relying on things being defined; I prefer to have configurable `#defines` be treated as booleans wherever possible.

I'd also rename the 'building' one just so they both share the same prefix, making it very obvious that they're related.

Reply:
I generally agree, but these are not options that should be exposed to users, so they shouldn't be already defined in any case. If you still think that checking their value is needed let me know

Reply:
It's got nothing to do with _user_ configurability, I just prefer doing preprocessor-based configuration with `#if` over `#ifdef`.

Reply:
> Edit: also because they don't have a "default" value, their value depends on the current build configuration and can't be changed

```
#ifndef TOML_SHARED_LIB
				#define TOML_SHARED_LIB 0
#endif
```

```
-DTOML_SHARED_LIB=1
```

Reply:
The difference between `#if` and `#ifdef` can bite you really hard even as the developer of a project, for example when porting between build systems. Accidentally forget to port one check over:
- use of `#ifdef` causes the compiler to effectively think the check returned false
- `#if` errors out since you didn't define it one way or another, and forces you to go back and check whether you actually did the check and defined the result.